### PR TITLE
ipc4: copier: set sink buffer format when sink format is set

### PR DESCRIPTION
**Pull request by RanderWang**
The original code only set sink format in copier but forget to set format of sink buffer. This will result to error when sink components use it.

### CODE REVIEW

File: src/audio/copier/copier.c [hunk 1]
```diff
@@ -1446,6 +1449,38 @@ static int copier_set_sink_fmt(struct comp_dev *dev, const void *data,
...
+		sink_c = buffer_acquire(sink);
+
+		/* only updated current used sink buffer */
+		sink_id = IPC4_SINK_QUEUE_ID(sink_c->id);
+		if (sink_id != sink_fmt->sink_id) {
+			buffer_release(sink_c);
+			continue;
+		}
+
+		sink_c->stream.channels = cd->out_fmt[sink_id].channels_count;
```
Comment (ranj063) [inline src/audio/copier/copier.c:1466]: why not use sink_fmt-> instead of cd->out_fmt[sink_id].?

File: src/audio/copier/copier.c [hunk 2]
```diff
@@ -1446,6 +1449,38 @@ static int copier_set_sink_fmt(struct comp_dev *dev, const void *data,
 							      &sink_fmt->sink_fmt, ipc4_gtw_none,
 							      ipc4_bidirection);
 
+	/* update corresponding sink format */
+	list_for_item(sink_list, &dev->bsink_list) {
```
Comment (ranj063) [inline src/audio/copier/copier.c:1453]: looks like this piece of code in already there in copier_params() on line 1360. Can you create a helper and reuse?

Comment (RanderWang) [reply src/audio/copier/copier.c:1449]: updated with a help funciton

File: src/audio/copier/copier.c [hunk 3]
```diff
@@ -1442,6 +1445,24 @@ static int copier_set_sink_fmt(struct comp_dev *dev, const void *data,
...
+		int sink_id;
+
+		sink = container_of(sink_list, struct comp_buffer, source_list);
+		sink_c = buffer_acquire(sink);
+
+		/* only updated current used sink buffer */
```
Comment (ranj063) [inline src/audio/copier/copier.c:1455]: nit-pick, this comment is not needed. the first comment to "update corresponding sink format" is enough

File: src/audio/copier/copier.c [hunk 4]
```diff
@@ -1297,6 +1297,27 @@ static void update_internal_comp(struct comp_dev *parent, struct comp_dev *child
 	child->direction = parent->direction;
 }
 
+static void update_buffer_format(struct comp_buffer __sparse_cache *buf_c,
+				 const struct ipc4_audio_format *in_fmt)
```
Comment (ranj063) [inline src/audio/copier/copier.c:1301]: another nit-pick, can we call this arg just fmt? in one case its input format and in another its output, so better to just keep this simple.

File: src/audio/copier/copier.c [hunk 5]
```diff
@@ -1297,6 +1297,27 @@ static void update_internal_comp(struct comp_dev *parent, struct comp_dev *child
 	child->direction = parent->direction;
 }
 
+static void update_buffer_format(struct comp_buffer __sparse_cache *buf_c,
```
Comment (ranj063) [inline src/audio/copier/copier.c:1300]: and you have a typo in the commit message. Maybe its better to say "Use helper function to update buffer format and avoid code duplication"

File: src/audio/copier/copier.c (hunk 4 above)
Comment (RanderWang) [reply src/audio/copier/copier.c:1301]: sure, thanks!

File: src/audio/copier/copier.c (hunk 3 above)
Comment (RanderWang) [reply src/audio/copier/copier.c:1455]: sure, thanks

File: src/audio/copier/copier.c (hunk 5 above)
Comment (RanderWang) [reply src/audio/copier/copier.c:1300]: thanks

**Review by ranj063 — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**